conformance/tests/mesh/httproute-redirect-port.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package meshtests
+
+import (
+	"testing"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/echo"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/roundtripper"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/pkg/features"
+)
+
+func init() {
+	MeshConformanceTests = append(MeshConformanceTests, MeshHTTPRouteRedirectPort)
+}
+
+var MeshHTTPRouteRedirectPort = suite.ConformanceTest{
```

Comment:
I think I missed this from the earlier conversations, but what's the plan to make these tests more sustainable + maintainable? Would it be viable to just have a simple script that did the copying and modification required for any tests that should be replicated for mesh? Is there a way to share the same test definitions for both?

Reply:
It would be a framework change. That is able to take the applicable test (probably using an enablement with a flag) and run it with a replaced parentRef and suite functions.

Reply:
Do we have any issues tracking that?

Reply:
We have https://github.com/kubernetes-sigs/gateway-api/issues/3581